### PR TITLE
Add Worker APIs to support DHP-1073 CSV Worker

### PR DESCRIPTION
**Pull request by DwayneJengSage**
https://sagebionetworks.jira.com/browse/DHP-1073

See https://github.com/Sage-Bionetworks/BridgeWorkerPlatform/pull/177

### CODE REVIEW

File: src/main/java/org/sagebionetworks/bridge/spring/controllers/AssessmentController.java [hunk 1]
```diff
@@ -95,7 +96,16 @@ public Assessment getAssessmentByGuid(@PathVariable String guid) {
         }
         return service.getAssessmentByGuid(appId, ownerId, guid);
     }
-    
+
+    @GetMapping("/v1/apps/{appId}/assessments/{guid}")
+    public Assessment getAssessmentByGuidForWorker(@PathVariable String appId, @PathVariable String guid) {
+        getAuthenticatedSession(WORKER);
+        if (SHARED_APP_ID.equals(appId)) {
+            throw new UnauthorizedException(SHARED_ASSESSMENTS_ERROR);
```
Comment (syoung-smallwisdom) [inline src/main/java/org/sagebionetworks/bridge/spring/controllers/AssessmentController.java:108]: So... the way things currently work, only shared assessments can be used by all organizations so the ARC assessments are all "shared". Won't those still need to be scored and have a CVS file created by the worker?

**Review by syoung-smallwisdom — COMMENTED** [review]
Confused about not allowing getting a shared assessment by appId.

File: src/main/java/org/sagebionetworks/bridge/spring/controllers/AssessmentController.java (hunk 1 above)
Comment (DwayneJengSage) [reply src/main/java/org/sagebionetworks/bridge/spring/controllers/AssessmentController.java:108]: The API doesn't allow shared assessments to be consistent with all the other APIs for assessments and shared assessments.

As per stand-up meeting today, the short term hack is to add fallback logic to the Worker to check for local assessments, then fall back to shared assessments. In 2024, we'll need to rework schedules and shared assessments to better fit how we actually use them.

**Review by DwayneJengSage — COMMENTED** [review]
Add getAssessmentConfigForWorker to unblock Nathaniel's stuff.

**Comment by DwayneJengSage** [comment]
@nategbrown9 This is updated with the assessment config API and unit tests.

**Review by nategbrown9 — APPROVED** [review]
Looks good.

**Review by syoung-smallwisdom — APPROVED** [review]
looks good

**Pull request merged**